From 6c9e44c5275345eebaad93106a9474174541b649 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Mon, 24 Oct 2022 09:01:31 +0800
Subject: [PATCH] sw64: fix compile error when CONFIG_DEBUG_PER_CPU_MAPS=y
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6c9e44c5275345eebaad93106a9474174541b649
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56QAM

--------------------------------

Signed-off-by: Mao Minkai <maominkai@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kvm/kvm-sw64.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/kvm/kvm-sw64.c b/arch/sw_64/kvm/kvm-sw64.c
index eb29583c1967..be5218234adf 100644
--- a/arch/sw_64/kvm/kvm-sw64.c
+++ b/arch/sw_64/kvm/kvm-sw64.c
@@ -614,7 +614,7 @@ int kvm_arch_vcpu_ioctl_run(struct kvm_vcpu *vcpu)
 			end = vcpu->kvm->arch.host_phys_addr + vcpu->kvm->arch.size;
 			nid = pfn_to_nid(PHYS_PFN(vcpu->kvm->arch.host_phys_addr));
 			if (pfn_to_nid(PHYS_PFN(end)) == nid)
-				set_cpus_allowed_ptr(vcpu->arch.tsk, node_to_cpumask_map[nid]);
+				set_cpus_allowed_ptr(vcpu->arch.tsk, cpumask_of_node(nid));
 		}
 #endif
 #else /* !CONFIG_KVM_MEMHOTPLUG */
-- 
2.33.0

